src/EditorFeatures/CSharpTest/MakeRefStruct/MakeRefStructTests.cs

```diff
@@ -0,0 +1,269 @@
+    }
+}
+
+ref struct S
+{
+    static Span<int> m;
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task PropInNotRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    Span<int>[||] M { get; }
+}
+";
+
+            var expected = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+ref struct S
+{
+    Span<int> M { get; }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task PropInNestedClassInsideNotRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    class C
+    {
+        Span<int>[||] M { get; }
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text, new TestParameters(s_parseOptions));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task PropStaticInRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+ref struct S
+{
+    static Span<int>[||] M { get; }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text, new TestParameters(s_parseOptions));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task PropStaticInNotRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    static Span<int>[||] M { get; }
+}
+";
+            var expected = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+ref struct S
+{
+    static Span<int> M { get; }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, parseOptions: s_parseOptions);
+        }
```

Comment:
do you have a fix-all test?  You returned the batch-fixer, so we should have a test for that.  Note: as @Neme12 has pointed out, the chance of needing fix-all in these cases is super low.  So we could even make this *not* batch-fixable and that would be 100% ok with me.

Reply:
Yep, I'll simply remove the batch-fixer. At first I decided not to add it, but found somewhere in the docs, that it is a bad practice for a new code-fixes.